extensions/wikia/LookupContribs/SpecialLookupContribs_helper.php

```diff
@@ -259,129 +232,51 @@ private function addWikiaInfo( $wikiaIds, &$userActivity ) {
 		}
 	}
 
-	private function getActivityCount( DatabaseBase $dbr, Array $where ) {
-		$res = $dbr->select(
-			'events',
-			[ 'count(distinct wiki_id) as num' ],
-			$where,
-			__METHOD__
-		);
-
-		$row = $dbr->fetchObject( $res );
-		return $row->num;
+	private function getUserActivityMemKey() {
+		return wfSharedMemcKey( $this->mUserId, 'data' );
 	}
 
-	private function getUserActivityMemKey( $addEditCount ) {
-		return wfSharedMemcKey( $this->mUserId, ( $addEditCount ? 'dataWithEdits' : 'data' ) );
-	}
+	private function orderData( $userActivity ) {
+		$data = $userActivity['data'];
 
-	private function orderData( &$data, $order, $edits ) {
-		$aTemp = [];
-		$aMatches = [];
-		if ( !$order ) {
-			$order = ( $edits ? 'edits:desc' : 'lastedit:desc' );
+		if ( empty( $data ) || !is_array( $data ) ) {
+			return $userActivity;
 		}
-		if ( isset( $data['data'] ) && is_array( $data['data'] ) ) {
 
-			// order by title
-			if ( preg_match( '/^title:(asc|desc)$/', $order, $aMatches ) ) {
-				foreach ( $data['data'] as $aItem ) {
-					$aTemp[$aItem['title']] = $aItem;
-				}
-			}
+		usort( $data, [ $this, 'sortUserActivity' ] );
 
-			// order by url
-			elseif ( preg_match( '/^url:(asc|desc)$/', $order, $aMatches ) ) {
-				foreach ( $data['data'] as $aItem ) {
-					$aTemp[$aItem['url']] = $aItem;
-				}
-			}
-
-			// order by last edit
-			elseif ( preg_match( '/^lastedit:(asc|desc)$/', $order, $aMatches ) ) {
-				foreach ( $data['data'] as $aItem ) {
-					// added URL part since last edits are ints and might not be unique
```

Comment:
The only reason the URL was combined with `last_edit` (and `editcount` below) is because of the extremely awkward way this sorting was being done.

I'm assuming the author only knew how to sort on an array's key or an array with simple scalar values.  To sort based on what they knew, they built a new temporary array `$aTemp` who's keys were a value from the complex (array of arrays) data structure, e.g.:
```php
$data = [
    [ 'url' => 'http://abc', 'title' => 'abc', 'last_edit' => '20150101' ],
    [ 'url' => 'http://xyz', 'title' => 'xzy', 'last_edit' => '20150102' ],
    ...
];
```
becomes:
```php
$aTemp = [
    '20150101' =>  [ 'url' => 'http://abc', 'title' => 'abc', 'last_edit' => '20150101' ],
    '20150102' => [ 'url' => 'http://xyz', 'title' => 'xzy', 'last_edit' => '20150102' ],
]
```
However, some of these values were identical (e.g. multiple items could have the same `last_edit` date), in which case each next identical value would overwrite the last in `$aTemp`. To prevent this and include all values in `$aTemp`, they appended something that would make it unique, e.g. the URL.

There is no reason that a set of items to sort need to be unique.  By using `usort` one can do the right thing which is return `0` in the sort function to tell sort the items are identical and no exchange is necessary.